pkg/controller/stackconfigpolicy/stackconfigpolicy.go

```diff
@@ -0,0 +1,401 @@
+		extractFunc: func(p *policyv1alpha1.StackConfigPolicy) policyv1alpha1.KibanaConfigPolicySpec {
+			return p.Spec.Kibana
+		},
+		mergeFunc: func(dst *policyv1alpha1.KibanaConfigPolicySpec, src policyv1alpha1.KibanaConfigPolicySpec, srcPolicy *policyv1alpha1.StackConfigPolicy) error {
+			if mergedPolicies == 0 {
+				// First policy: copy directly without merging/canonicalizing to avoid unnecessary differences
+				specCopy := src.DeepCopy()
+				// SecureSettings are aggregated by sSrcAggr
+				specCopy.SecureSettings = nil
+				*dst = *specCopy
+			} else {
+				var err error
+				if dst.Config, err = deepMergeConfig(dst.Config, src.Config); err != nil {
+					return err
+				}
+			}
+
+			sSrcAggr.aggregate(srcPolicy.Spec.Kibana.SecureSettings, srcPolicy)
+			mergedPolicies++
+			return nil
+		},
+	}
+
+	if err := merge(cfgPolicy, kbn, allPolicies, params.OperatorNamespace); err != nil {
+		return cfgPolicy, err
+	}
+	cfgPolicy.SecretSources = sSrcAggr.namespacedSecretSources
+	return cfgPolicy, nil
+}
+
+// doesPolicyMatchObject checks if the given StackConfigPolicy targets the given object.
+// A policy targets an object if both following conditions are met:
+// 1. The policy is in either the operator namespace or the same namespace as the object
+// 2. The policy's label selector matches the object's labels
+// Returns true if the policy targets the object, false otherwise, and an error if the label selector is invalid.
+func doesPolicyMatchObject(policy *policyv1alpha1.StackConfigPolicy, obj metav1.Object, operatorNamespace string) (bool, error) {
+	// Convert the label selector to a selector object
+	selector, err := metav1.LabelSelectorAsSelector(&policy.Spec.ResourceSelector)
+	if err != nil {
+		return false, err
+	}
+
+	// Check namespace restrictions; the policy must be in operator namespace or same namespace as the target object
+	if policy.Namespace != operatorNamespace && policy.Namespace != obj.GetNamespace() {
+		return false, nil
+	}
+
+	// Check if the label selector matches the Elasticsearch labels
+	if !selector.Matches(labels.Set(obj.GetLabels())) {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// deepMergeConfig merges the source Config into the destination Config using canonical configuration merging.
+// The merge is performed at the field level, with source values overriding destination values.
+// If src is nil, dst is returned unchanged. If dst is nil, it is initialized before merging.
+// Returns the merged config and any error occurred during config parsing or merging.
+func deepMergeConfig(dst *commonv1.Config, src *commonv1.Config) (*commonv1.Config, error) {
+	if src == nil {
+		return dst, nil
+	}
+
+	var dstCanonicalConfig *settings.CanonicalConfig
+	var err error
+	if dst == nil {
+		dst = &commonv1.Config{}
+		dstCanonicalConfig = settings.NewCanonicalConfig()
+	} else {
+		dstCanonicalConfig, err = settings.NewCanonicalConfigFrom(dst.DeepCopy().Data)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
```suggestion
	var err error
	if dst == nil {
		return src.DeepCopy(), nil
	}

	dstCanonicalConfig, err = settings.NewCanonicalConfigFrom(dst.DeepCopy().Data)
	if err != nil {
		return nil, err
	}
```

That also allows to get rid of the special casing for the single policy case at least for Kibana.